Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java

```diff
@@ -0,0 +1,68 @@
+package edu.asu.spring.quadriga.web.workbench;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.ModelAndView;
+
+import edu.asu.spring.quadriga.aspects.annotations.AccessPolicies;
+import edu.asu.spring.quadriga.aspects.annotations.CheckedElementType;
+import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
+import edu.asu.spring.quadriga.domain.factory.workbench.IPublicPageFactory;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+@Controller
+public class PublicPageController {
+
+    @Autowired
+    private IPublicPageFactory publicPageFactory;
+    
+    //This will be uncommented in QUAD-111
+    /*@Autowired
+    private IModifyPublicPageContentManager publicPageContentManager;*/
+
+    /**
+     * This method is called during the load of Public page settings form
+     *
+     * @return model - model object
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN,
+            RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/{projectid}/addpublicpage", method = RequestMethod.GET)
+    public ModelAndView publicPageSettingsForm(
+            @PathVariable("projectid") String projectid)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        ModelAndView model = new ModelAndView("auth/workbench/addpublicpage");
+        model.getModelMap().put("publicpage",
+                publicPageFactory.createPublicPageObject());
+        model.getModelMap().put("publicpageprojectid", projectid);
+        return model;
+    }
+
+    /**
+     * This method is used update the database with the information provided in the Public settings page
+     *
+     * @return json
+     */
+
+    @RequestMapping(method = RequestMethod.POST, value = "auth/workbench/{projectid}/addpublicpagesuccess")
+    public @ResponseBody ResponseEntity<String> addpublicpagesuccess(
+            @RequestParam("data") String data)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        if (data.isEmpty())
+            return null;
+        else
+            //Manager class will be called here
```

Comment:
as we talked earlier, please write this code here. Work with @TharaSridhar to agree on an interface that you can autowire, and then you should fill this class, as you know best how it works with the jsp page.